Sources/TimeStorage.swift

```diff
@@ -22,16 +22,34 @@ public enum TimeStoragePolicy {
 
 /// Handles saving and retrieving instances of `TimeFreeze` for quick retrieval
 public struct TimeStorage {
+    private let protection = NSLock()
```

Comment:
Sorry for my mistake.
As this is a value type, protection is inappropriate in this context.

Reply:
Protection has to be done outer caller context.